moya/doclinker.ts

```diff
@@ -0,0 +1,192 @@
+
+// Helpers
+
+function createComment(body: string) {
+  schedule(async () => {
+    await danger.github.api.issues.createComment({
+      owner: repo.owner.login,
+      repo: repo.name,
+      number: issue.number,
+      body: body
+    })
+  })
+}
+
+function issueIsQuestion(): boolean {
+  const labels = issue.labels
+  for (var i = 0; i < labels.length; i++) {
+    const label = labels[i]
+    if (label.name == "question") { return true }
+  }
+  return false
+}
+
+function titleIncludesAny(words: Set<string>): boolean {
+  if (titleWords.length == 0) { return false }
+  for (var i = 0; i < titleWords.length; i++) {
+    const titleWord = titleWords[i]
+    if (words.has(titleWord)) { return true }
+  }
+  return false
+}
+
+function addLinkToBody(text: string, link: string) {
+  commentBody.concat(`- [${text}](${link}) \n`)
+}
+
+function createCommentBody() {
+
+  // Always add Basic.md as a suggestion
+  addLinkToBody("Getting Started with Moya", basicURL)
+
+  if (titleIncludesAny(migrationWords)) {
+    addLinkToBody("Migration Guides", migrationURL)
+  }
+
+  if (titleIncludesAny(targetWords)) {
+    addLinkToBody("Targets", targetsURL)
+  }
+
+  if (titleIncludesAny(providerWords)) {
+    addLinkToBody("Providers", providerURL)
+  }
+
+  if (titleIncludesAny(endpointWords)) {
+    addLinkToBody("Endpoints", endpointsURL)
+  }
+
+  if (titleIncludesAny(multiTargetWords)) {
+    addLinkToBody("MultiTarget", multiTargetURL)
+  }
+
+  if (titleIncludesAny(multipartWords)) {
+    addLinkToBody("MultipartUpload", multipartURL)
+  }
+
+  if (titleIncludesAny(encodingWords)) {
+    addLinkToBody("ParameterEncoding", encodingURL)
+  }
+
+  if (titleIncludesAny(responseWords)) {
+    addLinkToBody("Response", responseURL)
+  }
+
+  if (titleIncludesAny(moyaErrorWords)) {
+    addLinkToBody("MoyaError", moyaErrorURL)
+  }
+
+  if (titleIncludesAny(reactiveSwiftWords)) {
+    addLinkToBody("ReactiveSwift", reactiveSwiftURL)
+  }
+
+  if (titleIncludesAny(rxSwiftWords)) {
+    addLinkToBody("RxSwift", rxSwiftURL)
+  }
+
+  if (titleIncludesAny(threadingWords)) {
+    addLinkToBody("Threading", threadingURL)
+  }
+
+  if (titleIncludesAny(authenticationWords)) {
+    addLinkToBody("Authentication", authenticationURL)
+  }
+
+  if (titleIncludesAny(pluginWords)) {
+    addLinkToBody("Plugins", pluginsURL)
+  }
+
+  if (titleIncludesAny(authpluginWords)) {
+    addLinkToBody("AuthPlugin", authpluginURL)
+  }
+
+  if (titleIncludesAny(customPluginWords)) {
+    addLinkToBody("CustomPlugins", customPluginURL)
+  }
+}
+
+// Github
+
+const gh = danger.github as any;
+const issue = gh.issue;
+const username = issue.user.login
+const repo = gh.repository;
+
+const title = issue.title.toLowerCase()
+const titleWords = title.split(" ")
+
+const commentHeader = `
+  Hey @${username},
+  We think the following documentation may be helpful in resolving your question:
+`
+
+var commentBody = ""
```

Comment:
Typically in ES6, we use `let` instead of `var` for non-`const` variables (which, I know, is confusing).

Reply:
Omg that is confusing 😆